src/projects/EnsureThat/Enforcers/EnumArg.cs

```diff
@@ -37,5 +39,48 @@ public sealed class EnumArg
 
             return value;
         }
+
+        /// <summary>
+        /// Confirms that the <paramref name="value"/> is defined in the enum <typeparamref name="T"/>.
+        /// Supports `Flags` attribute.
+        /// </summary>
+        public T IsDefined<T>(T value, [InvokerParameterName] string paramName = null, OptsFn optsFn = null) where T : struct, Enum
+        {
+            var enumType = typeof(T);
+            bool isEnumDefined;
+            var hasFlags = enumType.GetTypeInfo().GetCustomAttributes<FlagsAttribute>(false).Any();
+            if (hasFlags)
+            {
+                var givenEnumValue = Convert.ToUInt64(value);
+                if (givenEnumValue == 0)
+                {
+                    isEnumDefined = Enum.IsDefined(enumType, value);
+                }
+                else
+                {
+                    var enumValues = Enum.GetValues(enumType)
+                        .Cast<object>()
+                        .Select(Convert.ToUInt64);
+                    var allEnumValues = enumValues.Aggregate<ulong, ulong>(0, (current, enumValue) => current | enumValue);
```

Comment:
Would this work for all Flags enums? Or only power-of-2 enums. 
Would an enum like

[Flags] enum Mine{
A = 4,
B = 5,
C = 8
}

acknowledge that 9 is defined?

Reply:
^ I think this, and the complexity of the method, is why the BCL didn't implement this :)

Reply:
I added test for this case and yes, it did acknowledge that 9 is defined. ;)

Reply:
I'm struggling to get my head in order right now, but if allowing non-power-of-2 enums I think we end up mapping to a bin-packing problem.

> Define enumeration constants in powers of two, that is, 1, 2, 4, 8, and so on. This means the individual flags in combined enumeration constants do not overlap

I think this is a very good defense that we can use to say "not supported" or "your fault if you did that", I just wanted us to be explicit in that since I know the compiler LETS you define non-power-of-2 enums and just throws a warning on it :-S

On the side, I do wonder at some level if we should detect for that when doing the one-time-enum-type startup? It would be O(n) where n is the number of values on the enum - the biggest argument _against_ doing the check would be that in the 99% case, it's just wasted cycles (but it only being done once per program execution I think perhaps mediates it, *shrug*).

Reply:
Ah, checking for power-of-two would be impossible because they can "name" composites. I think just saying "implement flags properly" is enough :)

Reply:
Yeah, I tried it, and found out something new (for me). If you have this enum:
```
[Flags]
        public enum TestFlagsEnum : byte
        {
            Bar = 1,
            Baz = 1 << 1,
            Foo = Bar | Baz
        }
```
Enum.IsDefined(TestFlagsEnum.Foo) returns true! This is new to me, I always thought that it wouldn't recognize the Foo, since it's a composite. I guess composites work in different way than I always thought. So yeah, checking for power-of-twos would throw an exception here, because Foo would be returned as actual value by  `Enum.GetValues(EnumType);`. If it wouldn't - this could be done. We could check if it's a composite or not, really, but not worth the time and cpu cycles it would waste for majority of people.